src/decorator/live-decorator.ts

```diff
@@ -1,76 +1,68 @@
-import { KalturaLivePlugin } from "../kaltura-live-plugin";
-import { getContribLogger } from "@playkit-js-contrib/common";
+import { KalturaLivePlugin } from '../kaltura-live-plugin';
+import { getContribLogger } from '@playkit-js-contrib/common';
 
 const HTTP_ERRORS = [1002, 1003, 3016, 3022];
-enum EventTypes {
-    ERROR = "error",
-    ENDED = "ended",
-    PLAYING = "playing",
-    ABORT = "abort",
-    PAUSED = "pause",
-    METADATA = "timedmetadata",
-    SEEKED = "seeked",
-};
 
 const logger = getContribLogger({
-    class: "KalturaLiveDecorator",
-    module: "kaltura-live-decorator"
+  class: 'KalturaLiveDecorator',
+  module: 'kaltura-live-decorator',
 });
 
-export class KalturaLiveEngineDecorator implements KalturaPlayerTypes.IEngineDecorator {
-    _plugin: KalturaLivePlugin;
-    _engine: any;
-    _dispatcher: Function;
+export class KalturaLiveEngineDecorator 
+  implements KalturaPlayerTypes.IEngineDecorator {
+  _plugin: KalturaLivePlugin;
+  _engine: any;
+  _dispatcher: Function;
+  _isDecoratorActive: boolean;
```

Comment:
Please change the name to `_hadError`, _isDecoratorActive could be confusing cause if you're not in live and error occur it will be true but the decorator won't be active